sdk/app_cpu1/common/drv/amds.c

```diff
@@ -123,4 +127,32 @@ void amds_get_counters(uint8_t port, uint32_t *V, uint32_t *C, uint32_t *T)
     }
 }
 
+int amds_get_trigger_to_edge_delay(uint8_t port, amds_channel_e channel, double *out)
+{
+    uint32_t base_addr = amds_port_to_base_addr(port);
+
+    if (base_addr == 0) {
+        // This means an invalid port argument was passed
+        cmd_resp_printf("AMDS: Invalid Port Argument\r\n");
+        return FAILURE;
+    } else {
```

Comment:
Hmm... nice to print a response, but can you move it to the command handler?

This amds driver function can easily be used by user C code, not only from the CLI. So, just return FAILURE here and handle the printf in the higher level code.

If you printf in other API functions, please move to higher level cmd handler code

Reply:
I just removed it... the cmd handler already checks if the port is valid and if it is not, it will print the INVALID_ARGUMENTS status.

https://github.com/Severson-Group/AMDC-Firmware/blob/0d27cc0f0cdc9f627493bc45ff0d6b0998f531a7/sdk/app_cpu1/common/sys/cmd/cmd_amds.c#L29-L38